preferencesfx/src/main/java/com/dlsc/preferencesfx/util/StorageHandler.java

```diff
@@ -122,10 +122,21 @@
   // asciidoctor Documentation - end::storageHandlerLoad[]
 
   /**
-   * Searches in the storage after a serialized ArrayList using the given key, deserializes and
-   * returns it as ObservableArrayList. When an ObservableList is deserialzed, Gson returns an
-   * ArrayList and needs to be wrapped into an ObservableArrayList. This is only needed for
-   * loading.
+   * Searches in the storage after a serialized Object using the given key, deserializes and returns
+   * it. Returns a default Object if nothing is found.
+   *
+   * @param <T>           the type of object returned by this method
+   * @param <U>           the type of the default object
+   * @param breadcrumb    the key which is used to search the serialized Object
+   * @param type          the type of object used for deserialization
+   * @param defaultObject the Object which will be returned if nothing is found
+   * @return the deserialized Object or the default Object if nothing is found
+   */
+  <T, U extends T> T loadObject(String breadcrumb, Class<T> type, U defaultObject);
```

Comment:
I am not 100% sure if having this `<U extends T>` is making sense. Of if it would be sufficient to use `<T>` for the defaultObject.

Same goes for loadObservableList

Any opinion??

Reply:
Mhmm.. Since I cannot think of an example that would require this I would suggest to start with `<T>`, I think we can always open it up more, but closing it down later would be a breaking change.